Content.Server/Construction/AnchorableSystem.cs

```diff
@@ -85,7 +89,26 @@ private void OnAnchorComplete(EntityUid uid, AnchorableComponent component, TryA
             // TODO: Anchoring snaps rn anyway!
             if (component.Snap)
             {
-                _transform.SetCoordinates(uid, xform.Coordinates.SnapToGrid(EntityManager, _mapManager));
+                var coordinates = xform.Coordinates.SnapToGrid(EntityManager, _mapManager);
+
+                if (_tagSystem.HasTag(uid, "Unstackable"))
+                {
+                    // Need to enforce the unstackable rules on anchoring also.
+                    var condition = new NoUnstackableInTile();
+                    if (!condition.Condition(args.User, coordinates, Direction.Invalid))
+                    {
+                        var message = condition.GenerateGuideEntry()?.Localization;
+                        if (message != null)
+                        {
+                            // Show the reason to the user:
+                            _popup.PopupEntity(Loc.GetString(message), uid, args.User);
+                        }
+
+                        return;
+                    }
+                }
+
+                _transform.SetCoordinates(uid, coordinates);
```

Comment:
Is setting the coordinates necessary? Transform updates are expensive and anchoring is already going to do it.

Reply:
I didn't add that line, that's the existing line where anchored items snap to the grid. I just lifted the variable out of the existing line.